api/instance/v1/instance_utils.go

```diff
@@ -86,3 +88,117 @@ func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*Update
 
 	return &UpdateIPResponse{IP: ipResponse.IP}, nil
 }
+
+// AttachVolumeRequest contains the parameters to attach a volume to a server
+type AttachVolumeRequest struct {
+	Zone     utils.Zone `json:"-"`
+	ServerID string     `json:"-"`
+	VolumeID string     `json:"-"`
+}
+
+// AttachVolumeResponse contains the updated server after attaching a volume
+type AttachVolumeResponse struct {
+	Server *Server `json:"-"`
+}
+
+// volumesToVolumeTemplates converts a map of *Volume to a map of *VolumeTemplate
+// so it can be used in a UpdateServer request
+func volumesToVolumeTemplates(volumes map[string]*Volume) map[string]*VolumeTemplate {
+	volumeTemplates := map[string]*VolumeTemplate{}
+	for key, volume := range volumes {
+		volumeTemplates[key] = &VolumeTemplate{ID: volume.ID, Name: volume.Name}
+	}
+	return volumeTemplates
+}
+
+// AttachVolume attaches a volume to a server
+func (s *API) AttachVolume(req *AttachVolumeRequest, opts ...scw.RequestOption) (*AttachVolumeResponse, error) {
+	// get server with volumes
+	getServerResponse, err := s.GetServer(&GetServerRequest{
+		Zone:     req.Zone,
+		ServerID: req.ServerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	volumes := getServerResponse.Server.Volumes
+
+	newVolumes := volumesToVolumeTemplates(volumes)
+
+	// add volume to volumes list
+	key := fmt.Sprintf("%d", len(volumes))
+	newVolumes[key] = &VolumeTemplate{
+		ID: req.VolumeID,
+		// name is ignored on this PATCH
+		Name: req.VolumeID,
+	}
+
+	// update server
+	updateServerResponse, err := s.UpdateServer(&UpdateServerRequest{
+		Zone:     req.Zone,
+		ServerID: req.ServerID,
+		Volumes:  &newVolumes,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &AttachVolumeResponse{Server: updateServerResponse.Server}, nil
+}
+
+// DetachVolumeRequest contains the parameters to detach a volume from a server
+type DetachVolumeRequest struct {
+	Zone     utils.Zone `json:"-"`
+	VolumeID string     `json:"-"`
+}
+
+// DetachVolumeResponse contains the updated server after detaching a volume
+type DetachVolumeResponse struct {
+	Server *Server `json:"-"`
+}
+
+// DetachVolume detaches a volume from a server
+func (s *API) DetachVolume(req *DetachVolumeRequest, opts ...scw.RequestOption) (*DetachVolumeResponse, error) {
+	// get volume
+	getVolumeResponse, err := s.GetVolume(&GetVolumeRequest{
+		Zone:     req.Zone,
+		VolumeID: req.VolumeID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if getVolumeResponse.Volume == nil || getVolumeResponse.Volume.Server == nil {
+		return nil, fmt.Errorf("expected volume to have value in response")
```

Comment:
we should slip in to different error.
// expected volume to have value in response

and 
// server should be attach to a server

Reply:
if if getVolumeResponse.Volume == nil {
return nil, fmt.Errorf("expected volume to have value in response")
}

if (getVolumeResponse.Volume.Server == nil) {
return nil, fmt.Errorf("server should be attach to a server")
}